File: Tydzien1/wpf1/Egzaminator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Inf04Tydzien1Zad5
{
    internal class Egzaminator
    {
        private static int LicznikEgzaminatorow {  get; set; }
        private int ID {  get; set; }
        public string Imie;
        public string Nazwisko;
        public enum Specjalizacje
        {
            INF02,
            INF03,
            INF04
        }
        public Specjalizacje Specjalizacja;

        public Egzaminator ( string imie, string nazwisko, Specjalizacje specjalizacja)
        {
            ID = Egzaminator.LicznikEgzaminatorow;
            Imie = imie;
            Nazwisko = nazwisko;
            Specjalizacja = specjalizacja;
            Egzaminator.LicznikEgzaminatorow += 1;
        }
        
        public static int getLicznikEgzaminatorow()
        {
            return Egzaminator.LicznikEgzaminatorow;
        }
        
        public int getID()
        {
            return this.ID;
        }



        public string FormatSpecjalizacji()
        {
            switch (this.Specjalizacja) 
            {
                case Specjalizacje.INF02:
                    return "INF.02";
                case Specjalizacje.INF03:
                    return "INF.03";
                case Specjalizacje.INF04:
                    return "INF.04";
                default:
                    return "-";
            }
        }

        override public string ToString()
        {
            StringBuilder formatId = new StringBuilder();
            int ileZerPrzed = 3 - this.ID.ToString().Length;
            for(int i = 0; i < ileZerPrzed; i++)
            {
                formatId.Append(0);
            }
            formatId.Append(this.ID);
            return $"[ID:{formatId.ToString()}] {this.Imie} {this.Nazwisko} | {this.FormatSpecjalizacji()}";
        }


    }
}

File: Tydzien1/java/Egzaminator.java
package org.example;

public class Egzaminator {
    private static int licznikEgzaminatorow;
    private final int ID;
    public String imie;
    public enum specjalizacje{
        INF02,
        INF03,
        INF04
    }
    public specjalizacje specjalizacja;

    public Egzaminator(String imie, specjalizacje specjalizacja) {
        this.ID = licznikEgzaminatorow;
        this.imie = imie;
        this.specjalizacja =specjalizacja;
        Egzaminator.licznikEgzaminatorow += 1;
    }

    public String formatSpecjalizacji(){
        switch (this.specjalizacja){
            case INF02 :
                return "INF.02";
            case INF03 :
                return "INF.03";
            case INF04 :
                return "INF.04";
        }
        return "-";
    }

    public static int getLicznikEgzaminatorow() {
        return licznikEgzaminatorow;
    }

    public int getID() {
        return ID;
    }

    @Override
    public String toString() {
        StringBuilder konstruktorID = new StringBuilder();
        int ileZerPrzed = 3 - String.valueOf(this.ID).length();
        for(int i = 0; i < ileZerPrzed; i++){
            konstruktorID.append("0");
        }
        konstruktorID.append(this.ID);
        String IDstring = konstruktorID.toString();
        return "[ID:" + IDstring + "] " + this.imie + " | " + this.formatSpecjalizacji();
    }
}
